### PR TITLE
Lost messages

### PR DESCRIPTION
**Pull request by GioviQ**
Receiver loses datagrams during processing time of events.
Added producer / consumer pattern to manage events to resolve the issue.

### CODE REVIEW

File: src/KNXLib/KnxReceiver.cs [hunk 1]
```diff
@@ -172,7 +206,16 @@ protected void ProcessCEMI(KnxDatagram datagram, byte[] cemi)
             }
             catch (Exception e)
             {
-                Logger.Error(ClassName, e);
```
Comment (abelsilva) [inline src/KNXLib/KnxReceiver.cs]: Hi GioviQ  

This really makes sense!
Thanks!

Can you just change the pull without this Logger.Error changes?
If you check, I added a Logger.Error(string id, Exception e) function that does this (based on your last pulls)

Comment (GioviQ) [inline src/KNXLib/KnxReceiver.cs]: I removed the extra Logger functions.
Thank you

Giovanni

**Comment by abelsilva** [comment]
Thanks

**Comment by abelsilva** [comment]
released new version with pr

**Pull request merged**